src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1397,104 +1408,115 @@ private int onDecodeConnect(
 
                 if (sessionExpiryInterval > 0)
                 {
-                    final int flags = connect.flags();
-                    final String topic = String.format(SESSION_WILDCARD_TOPIC_FORMAT, clientId);
-                    final int topicKey = topicKey(topic);
-                    final RouteFW route = resolveTarget(routeId, authorization, topic,  PUBLISH_ONLY);
-
-                    if (route != null)
+                    progress = onResolveSession(traceId, authorization, reasonCode, progress, clientIdentifier, connect, payload);
+                }
+                else
+                {
+                    if (reasonCode == SUCCESS)
                     {
-                        final long resolvedId = route.correlationId();
+                        doCancelConnectTimeoutIfNecessary();
+                        doEncodeConnack(traceId, authorization, reasonCode, clientIdentifier.value());
+
+                        connected = true;
+                        keepAlive = connect.keepAlive();
+                        keepAliveTimeout = Math.round(TimeUnit.SECONDS.toMillis(keepAlive) * 1.5);
+                        doSignalKeepAliveTimeoutIfNecessary();
+                        decoder = decodePacketType;
+                        progress = connect.limit();
+                    }
+                }
+            }
 
-                        final boolean willFlagSet = isSetWillFlag(flags);
+            if (reasonCode != SUCCESS)
+            {
+                doCancelConnectTimeoutIfNecessary();
+                doEncodeConnack(traceId, authorization, reasonCode, clientIdentifier.value());
+                doNetworkEnd(traceId, authorization);
+                progress = connect.limit();
+            }
```

Comment:
Does this need to be split into two sections?
  - `if (reasonCode != SUCCESS || sessionExpiryInterval == 0)` => cancel connect timeout + encode connack
  - `if (reasonCode != SUCCESS)` => doNetworkEnd + progress

For the second one, it is sufficient to call `onDecodeError` instead, causing the decoder to switch to ignore-all and for `doNetworkEnd` to be called implicitly?